### ISSUE TITLE
[GUI] GUI Plugins are rebuilt at each launch (on OSX)

### ISSUE THREAD

**Issue by JusteRaimbault** [openmole/openmole#352]
I try to run OpenMOLE on OSX 10.12 (Sierra) and GUI plugins are entirely rebuilt each time I launch.
What could be the reason for that and could there be a simple solution ?
Otherwise I'll give up mac, as I guess it's not the aim to fully support it.

**Comment by mathieuleclaire** [comment]
Do you change GUI code before launching ? If yes it is normal. Note that you should use the --unoptimizedJS to speed up the launching.

**Comment by AnonymousReviewer** [comment]
No, the strange thing is that it really happen at each launch, without touching anything in openMOLE code. What is the test to check if gui plugins need to be rebuilt ?
thanks for the unoptimizedJS tip, it should help gaining some time in any case !

**Comment by JusteRaimbault** [comment]
[LOL who is this anonymous guy commenting on the issue ? - github automatic login fail, sorry ^^ ]

**Comment by mathieuleclaire** [comment]
Well, each time you modify some scala-js part (plugin or not), the js files will be re-build. In details, the jsPluginDirectory hash is tested. If modified, the js is re-build. https://github.com/openmole/openmole/blob/dev/openmole/gui/server/org.openmole.gui.server.core/src/main/scala/org/openmole/gui/server/core/Utils.scala#L353:L354

**Comment by JusteRaimbault** [comment]
ok thanks ! then I'll try to do some tests to understand why it appears as changed everytime on osx

**Comment by mathieuleclaire** [comment]
It only appears on your mac ? If yes, you should try the result of the hash.

**Comment by JusteRaimbault** [comment]
yes only on the mac.. I'll check this when I be on it again
